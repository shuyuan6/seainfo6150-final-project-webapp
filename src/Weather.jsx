import React from 'react'
import { useEffect, useState } from "react";
import { isEmpty } from "lodash";
import styles from "./Weather.module.css";

const Weather = (props) => {
    const [fetchedData, setFetchedData] = useState([]);

    
    useEffect(() => {
        
        const fetchData = async () => {
            // performs a GET request
            const response = await fetch(
                "http://demo7298209.mockable.io/weather"
            );
            //console.log(response)
            const responseJson = await response.json();
            console.log("Log from weather component")
            console.log(responseJson)
            setFetchedData(responseJson);
            //setFetchedData(Object.values(responseJson));
        }
        
        if (fetchedData.length === 0) {
            console.log("Calling fetech data")
            fetchData();
        }
        
    }, [fetchedData]);
    
    let displayContent;

    if (!isEmpty(fetchedData)) {
        displayContent = (
        <div className = {styles.weather}> 
            <b >Weather forecast</b>&nbsp;&nbsp;&nbsp;&nbsp;{fetchedData.city} : {fetchedData.weather}
            </div>
        );
    } else {
        displayContent = (<div>Weather Data not available</div>);
    }

    return displayContent
}

export default Weather