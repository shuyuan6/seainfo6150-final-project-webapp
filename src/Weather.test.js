import React from "react";
import { render } from "@testing-library/react";
import Weather from "./Weather.jsx";

it("renders the Weather component correctly", () => {
  const { container } = render(<Weather />);
  expect(container).toMatchSnapshot();
});
