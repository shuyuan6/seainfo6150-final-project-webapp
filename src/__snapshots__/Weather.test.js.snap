// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Weather component correctly 1`] = `
<div>
  <div>
    Weather Data not available
  </div>
</div>
`;
